VideoWeb/VideoWeb/ClientApp/src/app/shared/mappers/participant-panel-model-mapper.ts

```diff
@@ -0,0 +1,51 @@
+import { ParticipantForUserResponse, RoomSummaryResponse } from 'src/app/services/clients/api-client';
+import { LinkedParticipantPanelModel } from 'src/app/waiting-space/models/linked-participant-panel-model';
+import { PanelModel } from 'src/app/waiting-space/models/panel-model-base';
+import { ParticipantPanelModel } from 'src/app/waiting-space/models/participant-panel-model';
+
+export class ParticipantPanelModelMapper {
+    mapFromParticipantUserResponse(pats: ParticipantForUserResponse[]): PanelModel[] {
+        const participants: PanelModel[] = [];
+        pats.forEach(x => {
+            if (x.linked_participants?.length > 0) {
+                if (!this.doesListHaveLinkedParticipant(participants, x)) {
+                    const linkedParticipants = this.mapLinkedParticipant(x, pats);
+                    const room = this.getVmrFromRoom(linkedParticipants, pats);
+                    const participant = LinkedParticipantPanelModel.fromListOfPanelModels(linkedParticipants, room?.label, room?.id);
+                    participants.push(participant);
+                }
+            } else {
+                const participant = new ParticipantPanelModel(x);
+                participants.push(participant);
+            }
+        });
+        return participants;
```

Comment:
do we only want to return the group once? this will return

group per user? not group per room

Reply:
yes, we only need a group to represent one row